src/frequenz/sdk/timeseries/_formula_engine/_formula_engine_pool.py

```diff
@@ -0,0 +1,117 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""A formula pool for helping with tracking running formula engines."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Type
+
+from frequenz.channels import Sender
+
+from ...actor import ChannelRegistry, ComponentMetricRequest
+from ...microgrid.component import ComponentMetricId
+from ._formula_generators._formula_generator import (
+    FormulaGenerator,
+    FormulaGeneratorConfig,
+)
+from ._resampled_formula_builder import ResampledFormulaBuilder
+
+if TYPE_CHECKING:
+    # Break circular import by enclosing these type hints in a `TYPE_CHECKING` block.
+    from .._formula_engine import (
+        FormulaReceiver,
+        FormulaReceiver3Phase,
+        _GenericEngine,
+        _GenericFormulaReceiver,
+    )
+
+
+class FormulaEnginePool:
+    """Creates and owns formula engines from string formulas, or formula generators.
+
+    If an engine already exist with a given name, it is reused instead.
+    """
+
+    def __init__(
+        self,
+        namespace: str,
+        channel_registry: ChannelRegistry,
+        resampler_subscription_sender: Sender[ComponentMetricRequest],
+    ) -> None:
+        """Create a new instance.
+
+        Args:
+            namespace: namespace to use with the data pipeline.
+            channel_registry: A channel registry instance shared with the resampling
+                actor.
+            resampler_subscription_sender: A sender for sending metric requests to the
+                resampling actor.
```

Comment:
I think the variable name is too long, can `subscription` be removed as it is not even mentioned in the documentation?

Reply:
This is the same name across all our data pipeline actors and logical_meter.  So it is a broader topic.  

For this PR,  it is probably more sensible to be consistent with existing code, but I agree and it would be good to track and change everywhere in one go.  Maybe you can bring this up in the next sdk meeting.

Reply:
makes sense. Yes, we can discuss it in the meeting